### PR TITLE
OKHTTP Update to v3.8.1

### PR DESCRIPTION
**Pull request by searchy2**
*Update OKHTTP to v3.8.1 and OKHTTP URLConnection to v3.0.0.

### CODE REVIEW

**Comment by searchy2** [comment]
Fixes PR #97

File: library/build.gradle [hunk 1]
```diff
@@ -2,8 +2,8 @@ apply plugin: 'com.android.library'
 
 dependencies {
     compile 'com.klinkerapps:logger:1.0.3'
-    compile 'com.squareup.okhttp:okhttp:2.5.0'
-    compile 'com.squareup.okhttp:okhttp-urlconnection:2.5.0'
+    compile 'com.squareup.okhttp3:okhttp:3.8.1'
+    compile 'com.squareup.okhttp3:okhttp-urlconnection:3.0.0'
```
Comment (klinker24) [inline library/build.gradle]: This needs to be the save 3.8.1 as okhttp, it was causing build errors in the sample app:

```groovy
compile 'com.squareup.okhttp3:okhttp-urlconnection:3.8.1'
```

Any reason why you didn't have it as 3.8.1?

Comment (searchy2) [inline library/build.gradle]: I thought v3.0.0 was the latest version. When I ran Lint "New Library Versions", no newer libraries were detected. Troubling! Change to v3.8.1

Comment (searchy2) [inline library/build.gradle]: Found the reason why. Apparently, Lint is really smart and the newest version of URLConnection that works with the code I submitted is actually v3.0.0. Any newer version seems to have removed HttpURLConnectionImpl so any usages of that method fails.

**Comment by searchy2** [comment]
Requested changes made. Thanks for reviewing!

**Comment by klinker24** [comment]
This doesn't seem to have worked as well as planned and I am not really sure what the difference is. If I had to guess, something internally changed with OkHttp and I don't know enough to know why it would break the implementation. I have had too many people come to me complaining that it is broken for it to be a coincidence, unfortunately.

I am going to revert our changes. I am not really worried about having conflicting versions of OkHttp with this library, because that isn't really the case. OkHttp3 is a completely different package than the OkHttp2 that we had been using, so anyone is more than welcome to use OkHttp3 in their projects, along with using this library. There won't be any risk involved in that.

**Comment by searchy2** [comment]
`This doesn't seem to have worked as well as planned and I am not really sure what the difference is. If I had to guess, something internally changed with OkHttp and I don't know enough to know why it would break the implementation. I have had too many people come to me complaining that it is broken for it to be a coincidence, unfortunately.`
Can you share a bit more information about the complaints? I would like to know if the new OKHTTP implementation is breaking in people's projects or on specific devices. Thanks!

**Comment by klinker24** [comment]
Users complaining that they can't send and receive. The build is fine, it isn't breaking on the developer side.

**Comment by searchy2** [comment]
That's concerning. Did you see any patterns in affected users? Maybe only users below a certain version of Android were affected?

**Comment by E3V3A** [comment]
`Users complaining that they can't send and receive. The build is fine, it isn't breaking on the developer side.`
Some more info would be useful. AOS and device versions and perhaps some logcats.

**Pull request merged**